rsocket/RSocketRequester.cpp

```diff
@@ -23,8 +23,15 @@ RSocketRequester::~RSocketRequester() {
   VLOG(1) << "Destroying RSocketRequester";
 }
 
-void RSocketRequester::setEventBase(folly::EventBase* evb) {
-  eventBase_ = evb;
+bool RSocketRequester::canReplaceEventBase() {
+  return eventBaseIsReplaceable_;
+}
+
+bool RSocketRequester::setEventBase(folly::EventBase* evb) {
+  if (canReplaceEventBase()) {
```

Comment:
Calls to RSocketRequester aren't synchronized, `eventBaseIsReplaceable_` and checks which depend on it need to be protected by a mutex

Reply:
Even making it atomic would not help?

Reply:
Which thread would setEventBase() be called on? I feel like the multithreading complexity of RSocket is already pretty high.

Reply:
@phoad  Both `eventBaseIsReplaceable_` and access to the underlying `eventBase_` need to be protected in the same critical section for them to be synchronized, and to prevent enqueuing on the `eventBase_` then having it be replaced